core/src/main/java/org/apache/iceberg/jdbc/JdbcCatalog.java

```diff
@@ -250,10 +266,114 @@ public void setConf(Configuration conf) {
     this.conf = conf;
   }
 
+  private boolean insertProperties(Namespace namespace, Map<String, String> properties) {
+    String namespaceName = JdbcUtil.namespaceToString(namespace);
+
+    try {
+      int insertedRecords = connections.run(conn -> {
+        StringBuilder sqlStatement = new StringBuilder(JdbcUtil.INSERT_NAMESPACE_PROPERTIES_SQL);
+
+        for (int i = 0; i < properties.size(); i++) {
+          if (i != 0) {
+            sqlStatement.append(", ");
+          }
+          sqlStatement.append(JdbcUtil.INSERT_PROPERTIES_VALUES_BASE);
+        }
+
+        try (PreparedStatement sql = conn.prepareStatement(sqlStatement.toString())) {
+          int rowIndex = 0;
+          for (Map.Entry<String, String> keyValue : properties.entrySet()) {
+            sql.setString(rowIndex + 1, catalogName);
+            sql.setString(rowIndex + 2, namespaceName);
+            sql.setString(rowIndex + 3, keyValue.getKey());
+            sql.setString(rowIndex + 4, keyValue.getValue());
+            rowIndex += 4;
+          }
+          return sql.executeUpdate();
+        }
+      });
+
+      if (insertedRecords == properties.size()) {
+        LOG.debug("Successfully committed to new namespace: {}", namespaceName);
+        return true;
+      } else {
+        throw new CommitFailedException("Failed to insertProperties to namespace %s in catalog %s", namespaceName,
```

Comment:
nit: prefer to not use code but human friendly language in error message, like "Failed to insert properties to namespace %s in catalog %s"

Reply:
+1

Reply:
I don't think that this is the correct exception to use. This is used when a table commit fails and its documentation states that it is the "Exception raised when a commit fails because of out of date metadata". This is the exception used to trigger retries when committing. It is probably best to use a more generic exception instead since this should not trigger a retry.

I might throw `IllegalStateException` here because this is an unexpected situation. `executeUpdate` will return the number of affected rows or 0 for statements that don't have rows. There is no return value for exceptional cases like the operation failing. Those are thrown as exceptions. Seeing this code path would be a really surprising result.

Reply:
I see the same exception in createTable
```
if (insertRecord == 1) {
      LOG.debug("Successfully committed to new table: {}", tableIdentifier);
    } else {
      throw new CommitFailedException("Failed to create table %s in catalog %s", tableIdentifier, catalogName);
    }
``` 
I can certainly change it to `IllegalStateException`